tests/unit/test_awsrequest.py

```diff
@@ -97,83 +97,6 @@ def tell(self):
         return self._stream.tell()
 
 
-class TestAWSRequest(unittest.TestCase):
-
-    def setUp(self):
-        self.tempdir = tempfile.mkdtemp()
-        self.request = AWSRequest(url='http://example.com')
-        self.prepared_request = self.request.prepare()
-        self.filename = os.path.join(self.tempdir, 'foo')
-
-    def tearDown(self):
-        shutil.rmtree(self.tempdir)
-
-    def test_should_reset_stream(self):
-        with open(self.filename, 'wb') as f:
-            f.write(b'foobarbaz')
-        with open(self.filename, 'rb') as body:
-            self.prepared_request.body = body
-
-            # Now pretend we try to send the request.
-            # This means that we read the body:
-            body.read()
-            # And create a response object that indicates
-            # a redirect.
-            fake_response = Mock()
-            fake_response.status_code = 307
-
-            # Then requests calls our reset_stream hook.
-            self.prepared_request.reset_stream_on_redirect(fake_response)
-
-            # The stream should now be reset.
-            self.assertEqual(body.tell(), 0)
-
-    def test_cannot_reset_stream_raises_error(self):
```

Comment:
In general, it seems like the coverage for `awsrequest` via `tests/unit/test_awsrequest.py` dropped with this PR.  It seems like we should at least keep the coverage we had before, and preferably raise the coverage given we're making changes to that file.

Reply:
Added a bunch of tests and fixed some of these tests.